File: sdks/ruby/rest/lib/Brume/types/webhook_delivery_list_response.rb
# frozen_string_literal: true

module Brume
  module Types
    # `GET /v1/webhooks/:id/deliveries` success response.
    class WebhookDeliveryListResponse < Internal::Types::Model
      field :deliveries, -> { Internal::Types::Array[Brume::Types::WebhookDeliveryItem] }, optional: false, nullable: false
    end
  end
end

File: sdks/php/rest/src/Types/WebhookDeliveryListResponse.php
<?php

namespace Brume\Rest\Types;

use Brume\Rest\Core\Json\JsonSerializableType;
use Brume\Rest\Core\Json\JsonProperty;
use Brume\Rest\Core\Types\ArrayType;

/**
 * `GET /v1/webhooks/:id/deliveries` success response.
 */
class WebhookDeliveryListResponse extends JsonSerializableType
{
    /**
     * @var array<WebhookDeliveryItem> $deliveries
     */
    #[JsonProperty('deliveries'), ArrayType([WebhookDeliveryItem::class])]
    public array $deliveries;

    /**
     * @param array{
     *   deliveries: array<WebhookDeliveryItem>,
     * } $values
     */
    public function __construct(
        array $values,
    ) {
        $this->deliveries = $values['deliveries'];
    }

    /**
     * @return string
     */
    public function __toString(): string
    {
        return $this->toJson();
    }
}
